_posts/2019-07-12-mtls-with-nginx.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: post
+current: post
+cover: datacenter.png
+title: Configuring Nginx with client certificate authentication (mTLS)
+date: 2019-07-12 12:00:00
+category: tutorials
+author: Viktor Petersson
+tags: [Open Source]
+class: post-template
+---
+
+#  Configuring Nginx with client certificate authentication (mTLS)
+
+**Required Skill Level**: Medium to Expert
+
+One of the cornerstones of Zero Trust Networking is Mutual TLS (known as mTLS). In simple terms, this means that each client is required to present a certificate to talk to the server. This is different compared to how your client (e.g. your web browser) only verifies the identity of the server. The client itself does not need to present any identity. The identity piece is normally solved using some kind of credential (such as a username/password or API token).
+
+By replacing credentials with certificates, we are able to significantly improve the security (in particular with short-lived certificates, like the ones we offer), while also making the implementation *easier* (as we remove the need for API key/credential management).
+
+In this article we will make this all more concrete and create a sample implementation. The sample installation will consist of a simple Python appserver, with an Nginx reverse proxy in front of it. Nginx will reject all connections without a valid certificate, and the appserver will then compare the certificate to a whitelist of devices that are allowed to talk to the server. Additionally, we also need a secondary device that we will use to speak to server.
```

Comment:
In this article we will make this all more concrete by creating a sample implementation. This will [...]